pkg/fanal/analyzer/language/golang/mod/mod.go

```diff
@@ -151,7 +165,11 @@ func (a *gomodAnalyzer) fillAdditionalData(apps []types.Application) error {
 			}
 
 			// e.g. $GOPATH/pkg/mod/github.com/aquasecurity/go-dep-parser@v1.0.0
-			modDir := filepath.Join(modPath, fmt.Sprintf("%s@%s", normalizeModName(lib.Name), lib.Version))
+			modDirSuffix := fmt.Sprintf("@%s", lib.Version)
+			if modPath == vendorPath {
+				modDirSuffix = ""
+			}
+			modDir := filepath.Join(modPath, fmt.Sprintf("%s%s", normalizeModName(lib.Name), modDirSuffix))
```

Comment:
I think it will be easier to understand this way.
```suggestion
			// Package dir from `vendor` dir doesn't have version suffix.
			modDirName := normalizeModName(lib.Name)
			if modPath != vendorPath {
				// Add version suffix for packages from $$GOPATH
				// e.g. $GOPATH/pkg/mod/github.com/aquasecurity/go-dep-parser@v1.0.0
				modDirName = fmt.Sprintf("%s@%s", modDirName, lib.Version)
			}
			modDir := filepath.Join(modPath, modDirName)
```